crates/config/src/revive.rs

```diff
@@ -0,0 +1,44 @@
+use foundry_compilers::{multi::MultiCompilerLanguage, ProjectPathsConfig};
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+use crate::Config;
+
+/// Filename for Revive cache
+pub const REVIVE_SOLIDITY_FILES_CACHE_FILENAME: &str = "revive-solidity-files-cache.json";
+
+/// Directory for Revive artifacts
+pub const REVIVE_ARTIFACTS_DIR: &str = "revive-out";
+
+pub const CONTRACT_SIZE_LIMIT: usize = 250_000;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
+/// Revive Config
+pub struct ReviveConfig {
+    /// Enable compilation using revive
+    pub revive_compile: bool,
+    /// The revive path
+    pub revive_path: Option<PathBuf>,
+    /// solc path to use along the revive compiler
+    pub solc_path: Option<PathBuf>,
+}
+
+impl ReviveConfig {
+    /// Returns the `ProjectPathsConfig` sub set of the config.
+    pub fn project_paths(config: &Config) -> ProjectPathsConfig<MultiCompilerLanguage> {
+        let builder = ProjectPathsConfig::builder()
+            .cache(config.cache_path.join(REVIVE_SOLIDITY_FILES_CACHE_FILENAME))
+            .sources(&config.src)
+            .tests(&config.test)
+            .scripts(&config.script)
+            .artifacts(config.root.join(REVIVE_ARTIFACTS_DIR))
```

Comment:
I dont think hard coding the revive `REVIVE_ARTIFACTS_DIR` is a good idea because we are discounting the fact that the user can also want to specify a custom dir using `--out` which wont work when compiling with revive WDYT?

Reply:
I think this part comes from your PR. What do you propose instead? Do you want to use what is specified by the config, no matter if the compilation is done using solc or resolc? I think we may have issue with compilation cache.